olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/DataRequest.h

```diff
@@ -41,6 +42,7 @@ namespace read {
 class DATASERVICE_READ_API DataRequest final {
  public:
   /**
+   * @deprecated
```

Comment:
Please move down after return and write why it is deprecated and what users should use instead, which means basically lock version in constructor.